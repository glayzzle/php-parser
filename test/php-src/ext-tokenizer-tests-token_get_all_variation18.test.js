// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/tokenizer/tests/token_get_all_variation18.phpt
  it("Test token_get_all() function : usage variations - with HTML code", function () {
    expect(parser.parseCode("<?php\n/*\n * Testing token_get_all() with source string containing HTML code with PHP\n *   HTML tags are considered to be T_INLINE_HTML(311)\n*/\necho \"*** Testing token_get_all() : 'source' string with HTML tags ***\\n\";\n$source = '\n<html>\n<body>\n    Testing HTML\n</body>\n</html>\"\n<?php\n    echo \"php code with HTML\";\n?>';\nvar_dump( token_get_all($source));\necho \"Done\"\n?>")).toMatchSnapshot();
  });
});
