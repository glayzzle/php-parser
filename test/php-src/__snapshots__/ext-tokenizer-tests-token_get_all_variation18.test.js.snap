// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Test token_get_all() function : usage variations - with HTML code 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"*** Testing token_get_all() : 'source' string with HTML tags ***\\\\n\\"",
          "unicode": false,
          "value": "*** Testing token_get_all() : 'source' string with HTML tags ***
",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "source",
        },
        "operator": "=",
        "right": String {
          "isDoubleQuote": false,
          "kind": "string",
          "raw": "'
<html>
<body>
    Testing HTML
</body>
</html>\\"
<?php
    echo \\"php code with HTML\\";
?>'",
          "unicode": false,
          "value": "
<html>
<body>
    Testing HTML
</body>
</html>\\"
<?php
    echo \\"php code with HTML\\";
?>",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "source",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "token_get_all",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    Echo {
      "expressions": Array [
        String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"Done\\"",
          "unicode": false,
          "value": "Done",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
